src/Navigation/Search/index.js

```diff
@@ -1,18 +1,36 @@
 import { useSelector, useDispatch } from "react-redux";
-import { useState, useEffect } from "react";
-import { useLocation } from "react-router-dom";
+import { useEffect } from "react";
+import { useLocation, useHistory } from "react-router-dom";
 import { Wrapper, Input, Icon } from "./styled";
-import { fetchData, selectPath,setPath} from "./searchSlice";
+import { fetchData, selectInputValue, selectPath, setInputValue, setPath } from "./searchSlice";
 
 export const Search = () => {
     const location = useLocation();
     const dispatch = useDispatch();
     const path = useSelector(selectPath);
-    const [query, setQuery] = useState("");
+    const inputValue = useSelector(selectInputValue);
+    const history = useHistory();
+    const searchParams = new URLSearchParams(location.search);
+    const query = searchParams.get("search");
+
+    useEffect(() => {
+        if (!query) {
+            history.push(`${location.pathname}`);
+        }
+    }, [query]);
+
+    useEffect(() => {
+        history.push(`${location.pathname}?search=${inputValue}`);
+    }, [inputValue]);
```

Comment:
I have a feeling that we should handle history.push not i Search but in MovieList and PeopleList, using `if..else` or something otherwise it interferes with `?pate=number`